include/envoy/stream_info/stream_info.h

```diff
@@ -236,6 +236,11 @@ class StreamInfo {
    */
   virtual void setResponseCodeDetails(absl::string_view rc_details) PURE;
 
+  /**
+   * @param termination_details the termination details string to set for this connection.
```

Comment:
s/connection/stream here and below. 

Also, I have a more fundamental question/comment: it seems strange to me that we have termination details next to response code details. I get that we are using stream info for both connections/streams, can stack them, etc., but it seems like if the real intention here is connection level vs. stream level, should this be `setConnectionTerminationDetails`? Effectively, if the stream results in the connection being terminated I think this makes more sense, otherwise the details would be in responseCodeDetails (where we also have details for resets)? WDYT?

Reply:
Thanks for the comment, just to confirm something before updating the PR.

`setConnectionTerminationDetails` totally makes sense as the original intention is to use this in the network filter (e.g. RBAC) when closing a connection for access control.

In your other comment: `This (termination_details) should be emitted somewhere I would remove this for now.`, do you have suggestion where to put the field? maybe in the AccessLogCommon? thanks.

Reply:
> In your other comment: This (termination_details) should be emitted somewhere I would remove this for now., do you have suggestion where to put the field? maybe in the AccessLogCommon? thanks.

What I meant is you didn't add any code (unless I missed it) that emits that field in the TCP/HTTP gRPC access log implementation. Shouldn't you add that if you are going to add it to the proto?

Reply:
Ah, I see, this should be emit in `source/extensions/access_loggers/grpc/tcp_grpc_access_log_impl.cc`, sorry I missed that completely. Renamed to `setConnectionTerminationDetails` and reverted the API changes for now.